apps/glusterfs/health.go

```diff
@@ -0,0 +1,123 @@
+package glusterfs
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+
+	"github.com/boltdb/bolt"
+	"github.com/lpabon/godbc"
+)
+
+var clusterHealth = "ClusterHealth"
+
+type dbchecker struct {
+	CurrentTime time.Time
+}
+
+func NewHealthz() *api.Healthz {
+	health := &api.Healthz{}
+	health.NodesStatus = make([]api.NodeHealthStatus, 0)
+	return health
+}
+
+func (d *dbchecker) Marshal() ([]byte, error) {
+	var buffer bytes.Buffer
+	enc := gob.NewEncoder(&buffer)
+	err := enc.Encode(*d)
+
+	return buffer.Bytes(), err
+}
+
+func (d *dbchecker) Unmarshal(buffer []byte) error {
+	dec := gob.NewDecoder(bytes.NewReader(buffer))
+	return dec.Decode(d)
+}
+
+func (*dbchecker) BucketName() string {
+	return BOLTDB_BUCKET_DBATTRIBUTE
+}
+
+func (d *dbchecker) Write(tx *bolt.Tx) error {
+	godbc.Require(tx != nil)
+	return EntrySave(tx, d, clusterHealth)
+}
+
+func (d *dbchecker) Read(tx *bolt.Tx) error {
+	godbc.Require(tx != nil)
+	return EntryLoad(tx, d, clusterHealth)
+}
+
+func (a *App) Healthz(w http.ResponseWriter, r *http.Request) {
+	info := NewHealthz()
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	//db Read check
+	err := a.db.View(func(tx *bolt.Tx) error {
```

Comment:
All of the db checking logic should be encapsulated in it's own function call so that the actual rest endpoint function call is simply formatting and reporting on the status of the various checks but not doing all of them directly.

Reply:
done